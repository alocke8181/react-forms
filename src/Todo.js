import React, {useState} from "react";

const Todo = ({id, text, removeTodo}) =>{

    const handleRemove = (e)=>{
        e.preventDefault();
        removeTodo(id);
    };

    return(
        <li className="todo-item" id={id}>
            {text}
            <button className="todo-remove-button" onClick={handleRemove}>X</button>
        </li>
    )

}

export default Todo