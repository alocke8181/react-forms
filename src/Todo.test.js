import { render } from '@testing-library/react';
import Todo from './Todo';

test('smoke',()=>{
    render(<Todo id={1} text={'test'} removeTodo={console.log} />)
})

test('snapshot',()=>{
    const {asFragment} = render(<Todo id={1} text={'test'} removeTodo={console.log} />)
    expect(asFragment()).toMatchSnapshot();
})
