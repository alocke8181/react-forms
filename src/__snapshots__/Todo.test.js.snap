// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <li
    class="todo-item"
    id="1"
  >
    test
    <button
      class="todo-remove-button"
    >
      X
    </button>
  </li>
</DocumentFragment>
`;
